xarray/tests/test_backends.py

```diff
@@ -772,7 +772,7 @@ def find_and_validate_array(obj):
                     if isinstance(obj.array, np.ndarray):
                         assert isinstance(obj, indexing.NumpyIndexingAdapter)
                     elif isinstance(obj.array, dask_array_type):
-                        assert isinstance(obj, indexing.DaskIndexingAdapter)
+                        assert isinstance(obj, indexing.ChunkedIndexingAdapter)
```

Comment:
These adapters work around specific limitations for a type. It might not make sense to generalize this.

Reply:
So if a `dask.array` should still be wrapped in a `DaskIndexingAdapter`, what should a non-dask but chunked array (e.g. cubed) be wrapped with? I could add a `CubedIndexingAdapter` but so far I've avoided special-casing cubed.

Reply:
It depends on the indexing ops supported by cubed.

Also @tomwhite added a `ArrayApiIndexingAdapter` earlier so perhaps that's all that is needed.

Reply:
Ohh right okay yeah if `ArrayApiIndexingAdapter` also passes the tests then that would presumably be a better generalization.

Reply:
I've reverted this change - I think the `ArrayApiIndexingAdapter` was already handling this nicely for me.